File: examples/solv_ore_syz.rb
#
# jruby examples for jas.
# $Id$
#

require "examples/jas"

# Ore extension solvable polynomial example, Gomez-Torrecillas, 2003

p = PolyRing.new(QQ(),"x,y,z,t");
#is automatic: [one,x,y,z,t] = p.gens();

relations = [z, y,  y * z + x,
             t, y,  y * t  + y,
             t, z,  z * t - z
            ];

puts "relations: = [" + relations.join(", ") { |r| r.to_s } + "]";
puts;

rp = SolvPolyRing.new(QQ(), "x,y,z,t", PolyRing.lex, relations);
puts "SolvPolyRing: " + str(rp);
puts;

puts "gens =" + rp.gens().join(", ") { |r| r.to_s };
#is automatic: one,x,y,z,t = rp.gens();

#f1 = x**2 + y**2 + z**2 + t**2 + 1;
#puts "f1 = " +str(f1);

ff = [ x, y, z, t**2 - 1 ];
puts "ff = [" + ff.join(", ") { |r| r.to_s } + "]";
puts

ii = rp.ideal( "", ff );
puts "SolvableIdeal: " + str(ii);
puts;


rgl = ii.leftGB();
puts "seq left GB: " + str(rgl);
puts "isLeftGB: " + str(rgl.isLeftGB());
puts;

s = rgl.leftSyzygy();
puts "syzygy: " + str(s);
puts;

t = rgl.isLeftSyzygy(s);
puts "is syzygy: " + str(t);
puts "is syzygy: " + str(s.isLeftSyzygy(rgl));
puts;


rgr = ii.rightGB();
puts "seq right GB: " + str(rgr);
puts "isRightGB: " + str(rgl.isRightGB());
puts;

sr = rgr.rightSyzygy();
puts "syzygy: " + str(sr);
puts;

t = rgr.isRightSyzygy(sr);
puts "is syzygy: " + str(t);
puts "is syzygy: " + str(sr.isRightSyzygy(rgr));
puts;



# module syzy

sgl = s.leftGB();
puts "seq left module GB: sgl = " + str(sgl);
puts

sml = sgl.leftSyzygy();
puts "left syzygy: sml = " + str(sml);
puts;

puts "is left syzygy: " + str(sml.isLeftSyzygy(sgl));
puts;


sgr = sr.rightGB();
puts "seq right module GB: sgl = " + str(sgr);
puts

#startLog();
smr = sgr.rightSyzygy();
puts "right syzygy: smr = " + str(smr);
puts;

puts "is right syzygy: " + str(smr.isRightSyzygy(sgr));
puts;



smr = smr.rightGB();
puts "seq right module GB: sgl = " + str(smr);
puts

smr = smr.rightSyzygy();
puts "right syzygy: smr = " + str(smr);
puts;

smr = smr.rightSyzygy();
puts "right syzygy: smr = " + str(smr);
puts;

File: examples/solv_ore_syz.py
#
# jython examples for jas.
# $Id$
#

import sys;

from jas import PolyRing, SolvPolyRing, QQ
from jas import startLog, terminate 

# Ore extension solvable polynomial example, Gomez-Torrecillas, 2003

p = PolyRing(QQ(),"x,y,z,t");
#is automatic: [one,x,y,z,t] = p.gens();

relations = [z, y,  y * z + x,
             t, y,  y * t  + y,
             t, z,  z * t - z
            ];

print "relations: = " + str([ str(f) for f in relations ]);
print;

rp = SolvPolyRing(QQ(), "x,y,z,t", PolyRing.lex, relations);
print "SolvPolyRing: " + str(rp);
print;

print "gens =" + str([ str(f) for f in rp.gens() ]);
#is automatic: one,x,y,z,t = rp.gens();

#f1 = x**2 + y**2 + z**2 + t**2 + 1;
#print "f1 = " +str(f1);

ff = [ x, y, z, t**2 - 1 ];
print "ff = " + str([ str(f) for f in ff ]);
print

ii = rp.ideal( "", ff );
print "SolvableIdeal: " + str(ii);
print;


rgl = ii.leftGB();
print "seq left GB: " + str(rgl);
print "isLeftGB: " + str(rgl.isLeftGB());
print;

s = rgl.leftSyzygy();
print "syzygy: " + str(s);
print;

t = rgl.isLeftSyzygy(s);
print "is syzygy: " + str(t);
print "is syzygy: " + str(s.isLeftSyzygy(rgl));
print;


rgr = ii.rightGB();
print "seq right GB: " + str(rgr);
print "isRightGB: " + str(rgl.isRightGB());
print;

sr = rgr.rightSyzygy();
print "syzygy: " + str(sr);
print;

t = rgr.isRightSyzygy(sr);
print "is syzygy: " + str(t);
print "is syzygy: " + str(sr.isRightSyzygy(rgr));
print;


# module syzy

sgl = s.leftGB();
print "seq left module GB: sgl = " + str(sgl);
print

sml = sgl.leftSyzygy();
print "left syzygy: sml = " + str(sml);
print;

print "is left syzygy: " + str(sml.isLeftSyzygy(sgl));
print;


sgr = sr.rightGB();
print "seq right module GB: sgl = " + str(sgr);
print

#startLog();
smr = sgr.rightSyzygy();
print "right syzygy: smr = " + str(smr);
print;

print "is right syzygy: " + str(smr.isRightSyzygy(sgr));
print;

